crates/ra_ide_api/src/matching_brace.rs

```diff
@@ -14,9 +17,15 @@ pub fn matching_brace(file: &SourceFile, offset: TextUnit) -> Option<TextUnit> {
         })
         .next()?;
     let parent = brace_node.parent();
-    let matching_kind = BRACES[brace_idx ^ 1];
+    let brace_pos = brace_idx ^ 1;
+    let matching_kind = BRACES[brace_pos];
     let matching_node = parent.children_with_tokens().find(|node| node.kind() == matching_kind)?;
-    Some(matching_node.text_range().start())
+
+    match (brace_pos % 2) as i32 {
+        L_BRACE => Some(matching_node.text_range().start()),
+        R_BRACE => Some(matching_node.text_range().end()),
+        _ => None,
```

Comment:
How is this possible?

Reply:
it's because I want to use a match for more expressive, btw the function also returns an Option value too
you have any suggestion for this?

Reply:
You could use `unreachable!()`.